tests/SpecTests/ClientSideEncryptionSpecTest.php

```diff
@@ -1642,6 +1642,48 @@ static function (self $test, Client $setupClient, ClientEncryption $clientEncryp
         ];
     }
 
+    /**
+     * Prose test 15: On-demand AWS Credentials
+     *
+     * @see https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests#on-demand-aws-credentials
+     * @testWith [true]
+     *           [false]
+     */
+    public function testOnDemandAwsCredentials(bool $shouldSucceed): void
+    {
+        $hasCredentials = (getenv('AWS_ACCESS_KEY_ID') && getenv('AWS_SECRET_ACCESS_KEY'));
+
+        if ($hasCredentials !== $shouldSucceed) {
+            Assert::markTestSkipped(sprintf('AWS credentials %s available', $hasCredentials ? 'are' : 'are not'));
+        }
+
+        $keyVaultClient = static::createTestClient();
+
+        $clientEncryption = new ClientEncryption([
+            'keyVaultClient' => $keyVaultClient->getManager(),
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['aws' => (object) []],
+        ]);
+
+        $dataKeyOpts = [
+            'masterKey' => [
+                'region' => 'us-east-1',
+                'key' => 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0',
+            ],
+        ];
+
+        if (! $shouldSucceed) {
+            $this->expectException(AuthenticationException::class);
```

Comment:
This is more specific than some other assertions we make in this file, but I inferred it from the corresponding libmongoc test in https://github.com/mongodb/mongo-c-driver/commit/9c4f49f2e9e69d6ae61070095457fb742a375919#diff-d363a5539f2fbabfadf2d7cc6cdb96a203fb63f2caa31bae731c3e12d97b204cR5013.